### DISCUSSION TITLE
Regression: color space woes

### DISCUSSION THREAD

**Discussion by axeld** [ImageMagick/ImageMagick#4202] [Help]
I have a JPG image source.jpg, and want to resize this to an sRGB image target.jpg like this -- this is with ImageMagick 7.0.10-62 on Windows 10:

`$ magick.exe 'source.jpg[0]' -background white -alpha off -auto-orient -profile 'sRGB_v4_ICC_preference.icc' +profile '*' -colorspace sRGB -resize '150x150>' -strip 'jpg:target.jpg' `

The colors come out reversed. The same source will be converted correctly if I switch back to ImageMagic 6.9.3-7.
Here are the files used:
[source.jpg](https://user-images.githubusercontent.com/863699/107197199-e0895c80-69f3-11eb-9903-80a35d351f3f.jpg)
[target.jpg](https://user-images.githubusercontent.com/863699/107197202-e121f300-69f3-11eb-9bb7-1d8f74157ba8.jpg)
[sRGB_v4_ICC_preference.zip](https://github.com/ImageMagick/ImageMagick/files/5942360/sRGB_v4_ICC_preference.zip)

**Comment by snibgo** [comment]
According to "magick identify", your input has CMYK channels but an RGB ICC profile. So which is correct? If the channels are correct and the profile is wrong, you might strip the profile, then assign a CMYK profile, then convert to a sRGB profile.

**Comment by axeld** [comment]
The source was also created with ImageMagick, so the underlying problem is probably introduced there. In any case, the source can be viewed okay everywhere, it's just that it will be wrong after the conversion -- so it would at least be nice if ImageMagick could handle it okay, too.

Anyway, the commands to create the source:
`magick.exe original.tif[0] -strip -background white -alpha off -profile 'LStar-RGB-v4.icc' -quality 85.0 source.jpg`

[original.zip](https://github.com/ImageMagick/ImageMagick/files/5943029/original.zip)
[LStar-RGB-v4.zip](https://github.com/ImageMagick/ImageMagick/files/5943031/LStar-RGB-v4.zip)

If I replace LStar with the sRGB profile from above, the result is correct. The LStar profile most probably came from [https://www.colormanagement.org/en/workingspaces.html](https://www.colormanagement.org/en/workingspaces.html)

**Comment by snibgo** [comment]
> magick.exe original.tif[0] -strip -background white -alpha off -profile 'LStar-RGB-v4.icc' -quality 85.0 source.jpg

That's the problem. The input file 4503_style_SP800_Perspektive_gone.tif is CMYK, with a CMYK profile. So ...
```
-profile 'LStar-RGB-v4.icc' 
```
... would convert from that CMYK profile to a RGB profile. But you precede this with "-strip", which removes the CMYK profile, and ...
```
-profile 'LStar-RGB-v4.icc' 
```
... assigns the RGB profile to the CMYK image.

I suggest you remove the "-strip". If you really want to strip, put it _after_ converting to the new profile.

**Comment by axeld** [comment]
Duh, okay, thanks! That explains me being stupid, but not why the outcome looks okay if I choose the sRGB profile instead. I that just coincidence?

**Comment by snibgo** [comment]
> If I replace LStar with the sRGB profile from above, the result is correct.

> ... but not why the outcome looks okay if I choose the sRGB profile instead. Is that just coincidence?

I don't know. sRGB_v4_ICC_preference.icc is a RGB profile, so assigning that to the CMYK file 4503_style_SP800_Perspektive_gone.tif creates a problem. Different viewers may handle this problem differently, and one viewer may handle the two profiles differently.

**Comment by axeld** [comment]
Alright, I mean, it would be nice if ImageMagick could follow the same logic when encountering invalid images as everyone else (including earlier versions of it), but since this only hides issues with that image, I also wouldn't mind if you just close that ticket as "wontfix" :-)

**Comment by snibgo** [comment]
IM has a philosophy of doing what the user asks for, if it is possible, even if it is strange. It assumes users know what they are doing.

Assigning a RGB profile to a CMYK image is a valid thing to do, but it takes human intelligence to realise you didn't really want to do that.

**Comment by axeld** [comment]
Thanks for pointing out that I didn't bring that intelligence to the table here (well, in my defense, the command line has been built programmatically, but I wrote that program) ;-)

In any case, I understand that philosophy. However, it would be nice to align the handling of the outcome of such failure to how it was handled in version 6.9.3-7. If that's not worthwhile or easily possible, or was done on purpose, I'm fine, too. Thanks for your help, it's very appreciated!
